.pre-commit-config.yaml

```diff
@@ -0,0 +1,47 @@
+# See https://pre-commit.com/hooks.html for more hooks
```

Comment:
Thanks for setting up the hooks. Maybe devs from the other cores have an opinion for adding other hooks?
@athril @hsorby @alan-wu ?

Reply:
The hooks look good to me.